Community/Tdarr_Plugin_Soggys_NVENC_HEVC_CQV_Optimised_Bitrate.js

```diff
@@ -0,0 +1,220 @@
+  if (inputs.ffmpeg_preset === undefined) {
+    ffmpeg_preset = `slow`
+    response.infoLog += '☑Preset not set, defaulting to slow\n'
+  } else {
+    ffmpeg_preset = `${inputs.ffmpeg_preset}`
+    response.infoLog += `☑Preset set as ${inputs.ffmpeg_preset}\n`
+  }
+
+  //codec will be checked so it can be transcoded correctly
+  if (file.video_codec_name == 'h263') {
+    response.preset = `-c:v h263_cuvid`
+  } else if (file.video_codec_name == 'h264') {
+    if (file.ffProbeData.streams[0].profile != 'High 10') {
+      //Remove HW Decoding for High 10 Profile
+      response.preset = `-c:v h264_cuvid`
+    }
+  } else if (file.video_codec_name == 'mjpeg') {
+    response.preset = `-c:v mjpeg_cuvid`
+  } else if (file.video_codec_name == 'mpeg1') {
+    response.preset = `-c:v mpeg1_cuvid`
+  } else if (file.video_codec_name == 'mpeg2') {
+    response.preset = `-c:v mpeg2_cuvid`
+  } else if (file.video_codec_name == 'vc1') {
+    response.preset = `-c:v vc1_cuvid`
+  } else if (file.video_codec_name == 'vp8') {
+    response.preset = `-c:v vp8_cuvid`
+  } else if (file.video_codec_name == 'vp9') {
+    response.preset = `-c:v vp9_cuvid`
+  } else if (file.video_codec_name == 'hevc') {
+    if (file.ffProbeData.streams[0].profile != 'High 10') {
+      response.preset = `-c:v hevc_cuvid`
+    }
+  }
+
+  //Set Subtitle Var before adding encode cli
+  for (var i = 0; i < file.ffProbeData.streams.length; i++) {
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_name.toLowerCase() == 'mov_text' &&
+        file.ffProbeData.streams[i].codec_type.toLowerCase() == 'subtitle'
+      ) {
+        subcli = `-c:s srt`
+      }
+    } catch (err) {}
+    //mitigate TrueHD audio causing Too many packets error
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_name.toLowerCase() == 'truehd' ||
+        (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'dts' &&
+          file.ffProbeData.streams[i].profile.toLowerCase() == 'dts-hd ma') ||
+        (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'aac' &&
+          file.ffProbeData.streams[i].sample_rate.toLowerCase() == '44100' &&
+          file.ffProbeData.streams[i].codec_type.toLowerCase() == 'audio')
+      ) {
+        maxmux = ` -max_muxing_queue_size 9999`
+      }
+    } catch (err) {}
+    //mitigate errors due to embeded pictures
+    try {
+      if (
+        (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'png' ||
+          file.ffProbeData.streams[i].codec_name.toLowerCase() == 'bmp' ||
+          file.ffProbeData.streams[i].codec_name.toLowerCase() == 'mjpeg') &&
+        file.ffProbeData.streams[i].codec_type.toLowerCase() == 'video'
+      ) {
+        map = `-map 0:v:0 -map 0:a -map 0:s?`
+      }
+    } catch (err) {}
+  }
+
+  response.preset += `,${map} -dn -c:v hevc_nvenc -b:v 0 -preset ${ffmpeg_preset} -cq ${inputs.cqv} -rc-lookahead 32 -bf ${inputs.bframe} -a53cc 0 -c:a copy ${subcli}${maxmux} -pix_fmt p010le`
```

Comment:
Perhaps you should give the user the option if they want 10 bit or not, or at least mention it in the description.